doc/api/http2.md

```diff
@@ -227,6 +227,8 @@ process has completed.
 A reference to the [`net.Socket`][] or [`tls.TLSSocket`][] to which this
 `Http2Session` instance is bound.
 
+Directly reading from and writing to the socket is strongly discouraged.
```

Comment:
Likely need to take the time to explain why, but this could be a whole section on its own.